### PR TITLE
Basic mysql/kafka events and messages example.

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: eventuate-tram-examples-common/src/main/java/io/eventuate/tram/examples/basic/events/DomainEventDispatcherInitializer.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package io.eventuate.tram.examples.basic.events;
+
+import io.eventuate.tram.events.subscriber.DomainEventDispatcher;
+import io.micronaut.context.annotation.Context;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+@Context
+public class DomainEventDispatcherInitializer {
```
Comment (cer) [inline eventuate-tram-examples-common/src/main/java/io/eventuate/tram/examples/basic/events/DomainEventDispatcherInitializer.java]: This should be moved into the same module as where the Micronaut bean for DomainEventDispatcherFactory is defined.

And generalized to have `DomainEventDispatcher[]`

File: eventuate-tram-examples-jdbc-kafka/src/test/java/io/eventuate/tram/examples/basic/kafka/JdbcKafkaTramMessageTest.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package io.eventuate.tram.examples.basic.kafka;
+
+import io.eventuate.tram.examples.basic.messages.AbstractTramMessageTest;
+import io.micronaut.context.annotation.Property;
+import io.micronaut.test.annotation.MicronautTest;
+
+@MicronautTest
+//@Property(name = "micronaut.eventuate.tram.message.producer.jdbc.factory", value = "true")
```
Comment (cer) [inline eventuate-tram-examples-jdbc-kafka/src/test/java/io/eventuate/tram/examples/basic/kafka/JdbcKafkaTramMessageTest.java]: Please remove this @Property stuff

File: eventuate-tram-examples-jdbc-kafka/build.gradle [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        mavenBom 'io.micronaut:micronaut-bom:1.1.0'
+    }
+}
+
+dependencies {
+    testCompile project(":eventuate-tram-examples-common")
+
+    testCompile "io.eventuate.messaging.kafka:eventuate-messaging-kafka-consumer-micronaut:$eventuateMessagingKafkaVersion"
+    testCompile "io.eventuate.messaging.kafka:eventuate-messaging-kafka-common-micronaut:$eventuateMessagingKafkaVersion"
+    testCompile "io.eventuate.messaging.kafka:eventuate-messaging-kafka-basic-consumer-micronaut:$eventuateMessagingKafkaVersion"
```
Comment (cer) [inline eventuate-tram-examples-jdbc-kafka/build.gradle]: These dependencies seem very numerous and complicated.

On the consumer side there is all this:

Why depend on `eventuate-messaging-kafka-consumer-micronaut` and `eventuate-messaging-kafka-basic-consumer-micronaut`. I would have expected `eventuate-messaging-kafka-consumer-micronaut` to already depend on `eventuate-messaging-kafka-basic-consumer-micronaut`.

I would also have expected `eventuate-messaging-kafka-consumer-micronaut`  to depend on `eventuate-messaging-kafka-common-micronaut`

And, then I see this `eventuate-tram-consumer-kafka-micronaut` - surely this depends on all of the above.

And on the producer side there is:

    testCompile "io.eventuate.tram.core:eventuate-tram-jdbc-micronaut:$eventuateTramVersion"
    testCompile "io.eventuate.tram.core:eventuate-tram-producer-jdbc-micronaut:$eventuateTramVersion"

`eventuate-tram-producer-jdbc-micronaut` already depends upon `eventuate-tram-jdbc-micronaut`

File: eventuate-tram-examples-common/src/main/java/io/eventuate/tram/examples/basic/events/DomainEventDispatcherInitializer.java (hunk 1 above)
Comment (dartartem) [reply eventuate-tram-examples-common/src/main/java/io/eventuate/tram/examples/basic/events/DomainEventDispatcherInitializer.java]: yes, I remember, will do

File: eventuate-tram-examples-jdbc-kafka/src/test/java/io/eventuate/tram/examples/basic/kafka/JdbcKafkaTramMessageTest.java (hunk 2 above)
Comment (dartartem) [reply eventuate-tram-examples-jdbc-kafka/src/test/java/io/eventuate/tram/examples/basic/kafka/JdbcKafkaTramMessageTest.java]: ok

File: eventuate-tram-examples-jdbc-kafka/build.gradle (hunk 3 above)
Comment (dartartem) [reply eventuate-tram-examples-jdbc-kafka/build.gradle]: I will fix

**Pull request merged**